### PR TITLE
Implement certificate policy manager for machines

### PR DESCRIPTION
**Pull request by GabrielNagy**
Details are in the individual commits

Fixes UDENG-1056

### CODE REVIEW

File: internal/policies/certificate/certificate.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+		log.Debugf(ctx, "Certificate policy entry: %#v", entry)
+	}
+
+	var action string
+	log.Debugf(ctx, "Certificate policy value: %d", value)
+	if value&enrollFlag == enrollFlag {
+		action = "enroll"
+	} else {
+		action = "unenroll"
+	}
```
Comment (denisonbarbosa) [inline internal/policies/certificate/certificate.go:208]:
```suggestion
	log.Debugf(ctx, "Certificate policy value: %d", value)
	action := "unenroll"
	if value&enrollFlag == enrollFlag {
		action = "enroll"
	}
```

Comment (didrocks) [reply internal/policies/certificate/certificate.go:208]: \o/ thanks!

File: internal/policies/certificate/certificate_test.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		"User, autoenroll not supported": {isUser: true, entries: []entry.Entry{enrollEntry}},
+
+		// Error cases
+		"Error on autoenroll script failure": {autoenrollScriptError: true, entries: []entry.Entry{enrollEntry}, wantErr: true},
+		"Error on invalid autoenroll value":  {entries: []entry.Entry{{Key: "autoenroll", Value: "notanumber"}}, wantErr: true},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		name := name
```
Comment (denisonbarbosa) [inline internal/policies/certificate/certificate_test.go:72]: I think we don't need to shadow the name here

File: internal/policies/certificate/certificate_test.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+				certificate.WithShareDir(filepath.Join(tmpdir, "sharedir")),
+				certificate.WithCertAutoenrollCmd(autoenrollCmd),
+			)
+
+			err := m.ApplyPolicy(context.Background(), "keypress", !tc.isUser, !tc.isOffline, tc.entries)
+			if tc.wantErr {
+				require.Error(t, err, "ApplyPolicy should fail")
+			} else {
+				require.NoError(t, err, "ApplyPolicy should succeed")
+			}
```
Comment (denisonbarbosa) [inline internal/policies/certificate/certificate_test.go:98]:
```suggestion
			if tc.wantErr {
				require.Error(t, err, "ApplyPolicy should fail")
			}
			require.NoError(t, err, "ApplyPolicy should succeed")
```

**Review by denisonbarbosa — COMMENTED** [review]
Looks great! Well done!

**Pull request merged**